Dockerfile.builder

```diff
@@ -53,7 +53,4 @@ RUN apt-get update \
 # Install Go
 #
 RUN curl -sfL https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -xzC /usr/local && \
-        go get -d -u github.com/gordonklaus/ineffassign && \
-        cd /go/src/github.com/gordonklaus/ineffassign && \
-        git checkout -b 1003c8bd00dc2869cb5ca5282e6ce33834fed514 1003c8bd00dc2869cb5ca5282e6ce33834fed514 && \
-        go install
+        GO111MODULE=on go get github.com/gordonklaus/ineffassign@1003c8bd00dc2869cb5ca5282e6ce33834fed514
```

Comment:
we might need to update the docker file tag in this PR

Reply:
Good point. I only updated it for #10806 (`cilium-builder:2020-04-01`).

Reply:
Image is currently building and I updated the tag in `Dockerfile` to `cilium-builder:2020-04-02`. While at it I also noticed that we can reduce the image size a bit by running `go clean -cache -modcache` after `go install` to get rid of the build and module download caches.